### PR TITLE
SMI protobuf Update

### PR DESCRIPTION
**Pull request by dhruv0000**
Signed-off-by: dhruv0000 <patel.4@iitj.ac.in>

**Description**

This PR fixes #

**Notes for Reviewers**

**[Signed commits](../CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [ ] Yes, I signed my commits.

### CODE REVIEW

File: smi/smi.go [hunk 1]
```diff
@@ -26,8 +26,7 @@ var (
 
 type SmiTest struct {
 	id             string
-	adaptorVersion string
-	adaptorName    string
+	mesh           ServiceMesh
```
Comment (kumarabd) [inline smi/smi.go:29]: make it pointer to `ServiceMesh`

File: smi/smi.go [hunk 2]
```diff
@@ -68,13 +72,17 @@ func New(ctx context.Context, id string, version string, name string, client *ku
...
+		name:    name,
+		version: version,
+	}
+
 	test := &SmiTest{
 		ctx:            ctx,
 		id:             id,
-		adaptorVersion: version,
 		kubeClient:     client,
+		mesh:           mesh,
```
Comment (kumarabd) [inline smi/smi.go:84]: Rather than a new variable, define it right here as 
&ServiceMesh{
...
}

**Review by kumarabd — COMMENTED** [review]
Looks good, just some minor changes

**Comment by leecalcote** [comment]
When referring to service meshes by type, please use the SMP protos.

**Review by leecalcote — APPROVED** [review]
LGTM, assuming this is inheriting SMP protos from meshkit.

**Review by kumarabd — APPROVED** [review]

**Comment by kumarabd** [comment]
> LGTM, assuming this is inheriting SMP protos from meshkit.

There is a major challenge with this, Ill discuss in today's catchup @leecalcote ?

**Comment by leecalcote** [comment]
@kumarabd Meshery needs to use SMP. Yes, let's talk about this, if needed.

**Comment by kumarabd** [comment]
Yes @leecalcote , this was raised before the discussion. I've not pushed new changes since then, smp yes understood!

**Pull request closed without merging**